lib/filter/tests/test_filters_common.c

```diff
@@ -37,15 +37,14 @@
 #include "apphook.h"
 #include "plugin.h"
 #include "scratch-buffers.h"
+#include "libtest/msg_parse_lib.h"
 #include <criterion/criterion.h>
 
 #include <time.h>
 #include <string.h>
 #include <stdlib.h>
 #include <stdio.h>
 
-MsgFormatOptions parse_options;
-
```

Comment:
Wouldn't be better to keep away us from libtest?

(I mean you wouldn't get this multiple definition issue if you would not add `LIBTEST` to the `add_unit_test` calls.
So the commit message is about resolving an issue which is caused by your change within the very same patch.
I guess that the reason of having `parse_options` definition in `test_filters_common.c` is that we didn't want to depend on `libtest`. But I also think that you are know this... so maybe there is something _I_ do not know... Could you add a more detailed explanation on why are you doing this? :) )

Reply:
The original problem was with the autotools build, where criterion and libtest are not switchable, both are linked unconditionally. I'll fix this correctly along with the failing macOS job.

Reply:
Thanks for the note. I've removed the `parse_options` global from the msg-parse lib. I think it's better this way, because we had a lot of unit tests where `parse_options` was used without any sign of declaration, definition, and initialization.

Removing this global revealed that `init_and_load_syslogformat_module()` actually means `init_parse_options_and_load_syslogformat()`, which is a strange combination. This could be split and refactored further.